ext/standard/fsock.c

```diff
@@ -73,14 +73,27 @@ static void php_fsockopen_stream(INTERNAL_FUNCTION_PARAMETERS, int persistent)
 	}
 
 	/* prepare the timeout value for use */
+	if (timeout != -1.0 && (timeout >= 0.0 || timeout <= (double) PHP_TIMEOUT_ULL_MAX / 1000000.0)) {
```

Comment:
```suggestion
	if (timeout != -1.0 && !(timeout >= 0.0 && timeout <= (double) PHP_TIMEOUT_ULL_MAX / 1000000.0)) {
```

The error message also isn't technically correct anymore

Reply:
Also, please check if -1 behaves correctly now.

Reply:
I added a test

Reply:
Sorry, I missed that in the test file